allennlp/modules/transformer/attention_module.py

```diff
@@ -0,0 +1,617 @@
+import math
+from typing import Optional, Tuple, TYPE_CHECKING
+from dataclasses import dataclass
+import torch
+import torch.nn.functional as F
+
+from allennlp.common import FromParams
+from allennlp.common.checks import ConfigurationError
+from allennlp.modules.attention import Attention
+from allennlp.modules.transformer.transformer_module import TransformerModule
+from allennlp.modules.transformer.util import apply_mask, FloatT, IntT, BoolT
+
+if TYPE_CHECKING:
+    from transformers.configuration_utils import PretrainedConfig
+
+
+@dataclass
```

Comment:
It's probably better to use `NamedTuple` for these return types, but I'm okay with this for now.